### PR TITLE
Add tron topology_spread_constraints support to PaaSTA

### PR DESCRIPTION
**Pull request by nemacysts**
This adds support for configuring Tron-launched pods with a default Topology Spread Constraint (and node affinities) that will spread pods out across multiple AZs - otherwise, Karpenter will overwhelmingly favor a single AZ due to our config

I've double-checked that the feature flag works by running this code on
infrastage as well (as well as with the tests)

NOTE: there's a lot of test "changes", but that's mostly since I noticed
that we were inconsistently mocking load_system_paasta_config() and I wanted
to centralize things to make sure that the tests were doing the right thing.

### CODE REVIEW

**Comment by nemacysts** [comment]
requires https://github.com/Yelp/Tron/pull/1003 and https://github.com/Yelp/task_processing/pull/220

File: paasta_tools/tron_tools.py [hunk 1]
```diff
@@ -594,18 +597,35 @@ def get_node_selectors(self) -> Dict[str, str]:
...
             raw_selectors_to_requirements(
-                raw_selectors=self.config_dict.get("node_selectors", {}),  # type: ignore
+                raw_selectors=self.config_dict.get("node_selectors", {}),
             )
         )
+
+        # PAASTA-18198: To improve AZ balance with Karpenter, we temporarily allow specifying zone affinities per pool
+        pool_node_affinities = load_system_paasta_config().get_pool_node_affinities()
+        if pool_node_affinities and self.get_pool() in pool_node_affinities:
+            current_pool_node_affinities = pool_node_affinities[self.get_pool()]
```
Comment (EmanElsaban) [inline paasta_tools/tron_tools.py:620]: what does this dictionary returns?

Comment (nemacysts) [reply paasta_tools/tron_tools.py:620]: this is a puppet-defined dict (see `paasta_tools::public_config::pool_node_affinities` internally) that defines a default set of affinities for certain pools (speaking of which, we should probably add an entry for the tron pool :p):
```
  default:
    topology.kubernetes.io/zone:
      - us-west-2a
      - us-west-2b
```

Comment (EmanElsaban) [reply paasta_tools/tron_tools.py:620]: ahh i see it returns the azs affinities:
```
paasta_tools::public_config::pool_node_affinities:
  # pnw-prod default should only run on us-west-2a and us-west-2b (unless there's a deploy_whitelist override)
  default:
    topology.kubernetes.io/zone:
      - us-west-2a
      - us-west-2b
 ```

**Review by EmanElsaban — DISMISSED** [review]

File: paasta_tools/tron_tools.py (hunk 1 above)
Comment (EmanElsaban) [reply paasta_tools/tron_tools.py:620]: ahh sorry just saw your answer, yea makes sense

**Review by jfongatyelp — APPROVED** [review]
I was tricked by the 'big bounce' test failing but had missed that our test config enables tron TSCs as well 👍 So this shouldn't cause a big bounce til we enable that (and only affects runs on pools w/ constraints specified), sgtm!

**Pull request merged**